src/linker/Linker/ILParameterIndex.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Mono.Linker
+{
+	/// <summary>
+	/// Represents the index of arguments passed to a function in IL. Generally prefer to use <see cref="ILLink.Shared.SourceParameterIndex"/> when possible. See also <seealso cref="Mono.Linker.ParameterHelpers"/>.
+	/// </summary>
+	/// 
```

Comment:
Please add a comment describing that you're using enums as a way to separate source/IL indexes in the C# type system (took me a while to figure that out), and that they're actually used as numbers (relying on the fact that enum is really just an Int32).
Would also be nice to describe the numbering scheme here (IL parameter indexes are >=0 and for "instance" call the `0` is `this` and others are +1 .... and then something similar for source indexes).